src/ui/common/src/components/pages/metric/id/index.tsx

```diff
@@ -0,0 +1,462 @@
+  artifactName,
+  createdAt,
+}) => {
+  return (
+    <Box width="100%" display="flex" alignItems="center">
+      <FontAwesomeIcon
+        height="24px"
+        width="24px"
+        style={{ marginRight: '8px' }}
+        icon={faCircleCheck}
+        color={'green'}
+      />
+      <Typography variant="h4" component="div">
+        {artifactName}
+      </Typography>
+      {createdAt && (
+        <Typography marginTop="4px" variant="caption" component="div">
+          Created: {createdAt}
+        </Typography>
+      )}
+    </Box>
+  );
+};
+
+type MetricDetailsPageProps = {
+  user: UserProfile;
+  Layout?: React.FC<LayoutProps>;
+};
+
+const MetricDetailsPage: React.FC<MetricDetailsPageProps> = ({
+  user,
+  Layout = DefaultLayout,
+}) => {
+  const dispatch: AppDispatch = useDispatch();
+  const navigate = useNavigate();
+  const { workflowId, workflowDagResultId, metricOperatorId } = useParams();
+
+  const [inputsExpanded, setInputsExpanded] = useState<boolean>(true);
+  const [outputsExpanded, setOutputsExpanded] = useState<boolean>(true);
+
+  const workflow = useSelector((state: RootState) => state.workflowReducer);
+  const workflowDag = workflow.dagResults.find((currentDag) => {
+    return currentDag.id === workflowDagResultId;
+  });
+
+  const workflowDagId = workflowDag?.workflow_dag_id;
+  const operator = (workflow.dags[workflowDagId]?.operators ?? {})[
+    metricOperatorId
+  ];
+
+  // // Get the operatorSpec so that we can show more metadata about the operator.
+  // const operatorSpec = operator?.spec;
+  // console.log('operatorSpec: ', operatorSpec);
+
+  // const logs =
+  //   workflow.operatorResults[metricOperatorId]?.result?.exec_state?.user_logs ??
+  //   {};
+  // console.log('logs: ', logs);
+  // const operatorError =
+  //   workflow?.operatorResults[metricOperatorId]?.result?.exec_state?.error;
+  // console.log('operatorError', operatorError);
+
+  // // Get the execution state of the operator:
+  // // TODO: Show execution state in a badge component next to title of operator.
+  // const execState: ExecState =
+  //   workflow?.operatorResults[metricOperatorId]?.result?.exec_state;
+  // console.log('execState: ', execState);
+
+  const metricResult = workflow.operatorResults[metricOperatorId];
+
+  useEffect(() => {
+    // TODO: Update this to contain the name of the operator
+    document.title = 'Metric Details | Aqueduct';
+    dispatch(handleGetWorkflow({ apiKey: user.apiKey, workflowId }));
+  }, []);
+
+  useEffect(() => {
+    if (workflow.dagResults.length > 0) {
+      dispatch(
+        handleGetOperatorResults({
+          apiKey: user.apiKey,
+          workflowDagResultId,
+          operatorId: metricOperatorId,
+        })
+      );
+    }
+  }, [workflow.dagResults]);
+
+  // Set up different metric input types for rendering in the inputs list.
+  // TODO: transform/handle response from API and render these appropriately.
+  const mockMetricInputs = [
```

Comment:
Wanted to simulate each type of node and show their appropriate icons.

This kind of code is a good candidate for a storybook ui story. I have a small task to bring back storybook, so perhaps I can move it there in the not too distant future.